petab/v1/problem.py

```diff
@@ -1005,3 +1006,177 @@ def n_priors(self) -> int:
             return 0
 
         return self.parameter_df[OBJECTIVE_PRIOR_PARAMETERS].notna().sum()
+
+    def add_condition(self, id_: str, name: str = None, **kwargs):
```

Comment:
To match column headers/`C.py`? Not exactly the same due to underscores... but we could decide whether to go for consistent or context-specific IDs everywhere.
```suggestion
    def add_condition(self, condition_id: str, condition_name: str = None, **kwargs):
```

Reply:
Alternative: change all of these `add_*` methods to take `**kwargs` that are used to create a `pd.Series`, which is then validated and concatenated. `get_*_df` can be used to set the index. Then no table-specific code, and no need to redefine column names here?
- `if not isinstance(kwarg, str) and isinstance(kwarg, list): kwarg = PARAMETER_SEPARATOR.join(map(str, kwarg))`

Reply:
Yeah, I was struggling with what would be preferable. In `add_condition`, it feels redundant to prefix everything with `condition_`. Then again, it might be considered confusing if the arguments don't match the table columns. For me, the former felt more important. Also with regards to potentially introducing a proper object model, I think we'd want things to be more pythonic and less petaby.

I don't think the kwargs-solution would be very convenient. That would mean, you'd have write `add_observable(**{petab.OBSERVABLE_ID:"foo", petab.SIMULATION_CONDITION_ID: "bar"})`, I think then I rather directly go back to constructing dataframes.

Reply:
I'll leave it as is for. As discussed elsewhere, the v2 API is likely to change drastically overall where these points will be addressed.

Reply:
Alright, fine for v1, we can revisit v2 in case we change the column names there